inbm/dispatcher-agent/dispatcher/source/ubuntu_source_manager.py

```diff
@@ -98,11 +102,19 @@ def __init__(self) -> None:
 
     def add(self, parameters: ApplicationAddSourceParameters) -> None:
         """Adds a source file and optional GPG key to be used during Ubuntu application updates."""
-        # Step 1: Add key (Optional)
+        # Step 1: Verify gpg key uri from trusted repo list 
         if parameters.gpg_key_name and parameters.gpg_key_uri:
+            try:
+               url = parameters.gpg_key_uri
+               #URL slicing to remove the last segment (filename) from the URL 
+               source = url.value[:-(len(url.value.split('/')[-1]) + 1)]
+               verify_source(source=source, dispatcher_broker=Dispatcher_Broker)
```

Comment:
**Suggestion:** The variable 'Dispatcher_Broker' should be an instance of 'DispatcherBroker', but it seems to be used as a class. Make sure to create an instance of 'DispatcherBroker' and pass it to 'verify_source'. [possible issue]
**NOTE:** Ensure that the suggested code is comprehensive. This serves as an example of the code you could use. The user should verify it after accepting the suggestion.
```suggestion
        if parameters.gpg_key_name and parameters.gpg_key_uri:
            try:
               url = parameters.gpg_key_uri
               #URL slicing to remove the last segment (filename) from the URL 
               source = url.value[:-(len(url.value.split('/')[-1]) + 1)]
               dispatcher_broker_instance = DispatcherBroker()
               verify_source(source=source, dispatcher_broker=dispatcher_broker_instance)
            except DispatcherException as err:
               raise SourceError(f"Source Gpg key URI verification check failed: {err}")
            # Step 2: Add key (Optional)
            add_gpg_key(parameters.gpg_key_uri, parameters.gpg_key_name)
```